src/components/data-table/ha-data-table.ts

```diff
@@ -243,7 +243,8 @@ export class HaDataTable extends LitElement {
                   <search-input
                     .hass=${this.hass}
                     @value-changed=${this._handleSearchChange}
-                    .label=${this.searchLabel}
+                    .label=${this.searchLabel ??
+                    this.hass.localize("ui.components.data-table.search")}
```

Comment:
The default label should be in `<search-input>`, which already has a default label.  The problem is that it's not localized: [`.label=${this.label || "Search"}`](https://github.com/home-assistant/frontend/blob/13d686bd67769df62f8e182329e895b496af67e9/src/components/search-input.ts#L36).

Reply:
Yes I saw that, but I hesitated to put it there. 

If I add localization to "Search" in search-input I think it should need a new localize key (`ui.common.search`), which we do not currently have. 

Since the only known missing localization was from a ha-data-table I thought it better to use the existing search key from `ui.components.data-table.search`. 

Is it better to create a new key ui.common.search and use that key in search-input?

Should I create ui.common.search and link it to ui.components.data-table.search, so it does not require new translation work?

Ideally wanted to avoid creating new strings that needed to be translated, but I'm not sure the most correct option.

Reply:
Yeah poked around and search labels are kind of a mess...

> Should I create ui.common.search and link it to ui.components.data-table.search, so it does not require new translation work?

Yes, I would do that for now, and also:
1. Leave the key for now, but all uses of `ui.components.data-table.search` I think can be eliminated after that
2. There are 4 other keys with just "Search" as the text.  Those should either be changed to `ui.common.search` and removed, or just eliminated in favor of the default.

In other words, allow custom labels by the top originating element, but `undefined` should be handled only at the bottom in order to minimize repetitive code.

> Ideally wanted to avoid creating new strings that needed to be translated, but I'm not sure the most correct option.

Yeah we don't do as great a job in that department as we can IMO (as evidenced by the multiple duplicates).  For example, if you made the above changes, it would then make sense to make the switch in Lokalise then push back to the repo.

Reply:
Ok I cleaned up some of these places. Others like in ZHA I did not because I don't have that to test on my dev instance. 

> it would then make sense to make the switch in Lokalise then push back to the repo.

I'm not really sure what this means, haven't spent much time in Lokalize.